core/plugins/org.fusesource.ide.foundation.core/src/org/fusesource/ide/foundation/core/util/CamelUtils.java

```diff
@@ -183,4 +191,58 @@ public boolean visit(IResource resource) throws CoreException {
 		}
 		return files;
 	}
+	
+	/**
+	 * retrieves the camel design editor
+	 * 
+	 * @return
+	 */
+	private static IEditorPart getDiagramEditor() {
```

Comment:
We have that but I can't use it due to dependency loop issue. So I put it to the foundation core and it is private, so its not API.